target-app/Extensions/LabelExtension.swift

```diff
@@ -50,13 +51,20 @@ extension UILabel {
             self.numberOfLines = numberOfLines
             self.textAlignment = textAlignment
             self.font = UIFont.boldSystemFont(ofSize: fontSize)
-        case .secondary(let text, let textColor, let backgroundColor, let numberOfLines, let textAlignment, let fontSize):
+        case .secondary(let text, let textColor, let backgroundColor, let numberOfLines, let textAlignment, let fontSize, let isBold):
             self.text = text
             self.textColor = textColor
             self.backgroundColor = backgroundColor
             self.numberOfLines = numberOfLines
             self.textAlignment = textAlignment
-            self.font = UIFont.boldSystemFont(ofSize: fontSize)
+            let font: UIFont = {
+                if isBold == true {
+                    return UIFont.boldSystemFont(ofSize: fontSize)
+                } else {
+                    return self.font.withSize(fontSize)
+                }
```

Comment:
```suggestion
                if isBold {
                    return UIFont.boldSystemFont(ofSize: fontSize)
                } else {
                    return self.font.withSize(fontSize)
                }
```

Reply:
Is it ok that we are not using the same font when it is bold? Maybe we can add support for `UIFont` to update `UIFont.Weight` on the same instance.
This way you could have something like `self.font = font.withWeight(isBold ? .bold : .regular)`